homeassistant/components/huawei_lte/select.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Huawei LTE selects."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from functools import partial
+import logging
+
+from huawei_lte_api.enums.net import LTEBandEnum, NetworkBandEnum, NetworkModeEnum
+
+from homeassistant.components.select import (
+    DOMAIN as SELECT_DOMAIN,
+    SelectEntity,
+    SelectEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import UNDEFINED
+
+from . import HuaweiLteBaseEntityWithDevice
+from .const import DOMAIN, KEY_NET_NET_MODE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class HuaweiSelectEntityDescription(SelectEntityDescription):
+    """Class describing Huawei LTE select entities."""
+
+    setter_fn: Callable[[str], None] = lambda x: None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up from config entry."""
+    router = hass.data[DOMAIN].routers[config_entry.entry_id]
+    selects: list[Entity] = []
+
+    desc = HuaweiSelectEntityDescription(
```

Comment:
Just a suggestion, but maybe create a module global for the select entities and loop them over here for creation? Will avoid need to refactor in the future when new selects are added.

Reply:
I thought about it when working on this and in a sense agree. But I've taught myself to resist such refactorings until actually known that they'll be needed, an how. For this particular case it would seem future selects would fit in the way we'd do the refactoring at this point with the data at hand, but one cannot know for certain if some other things are called for at that point, and then we would have refactored twice.

Certainly not a biggie and I can do it either way though.

Reply:
Your call, this looks good to me :-)